presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -311,6 +311,13 @@ private boolean isColumnsEnumeratingTable(InformationSchemaTable table)
     }
 
     private <T> Optional<Set<String>> filterString(TupleDomain<T> constraint, T column)
+    {
+        return generateFilterFromTupleDomain(constraint, column)
+                .map(filteredString -> filteredString.stream()
+                        .filter(value -> !value.isEmpty()).collect(toImmutableSet()));
```

Comment:
This `.filter(value -> !value.isEmpty())` can be only called for columns that we known that cannot have empty values (catalog, schema, table, column) however `filterString` suggests it is generic. Maybe rename or `filterEmptyValues(filterString(...))`? So it would be used in place where column is known.